### PR TITLE
Load file when peaks are provided

### PR DESCRIPTION
**Pull request by Guisch**
fix https://github.com/katspaugh/wavesurfer.js/issues/1414

### CODE REVIEW

**Comment by TimSC** [comment]
This works pretty well for me! +1 for merge into main project.

**Comment by thijstriemstra** [comment]
thanks for the feedback @TimSC. I'm afraid it'll break stuff, or there's a better solution, your thoughts @katspaugh?

**Comment by entonbiba** [comment]
@thijstriemstra yea `this.tmpEvents.push(this.once('interaction', load));` removing that might cause issues,  the reason that code is there is to make sure the peaks are loaded first before user interacts with the audio.

Would it hurt to change it from `"interaction"` to `"ready"` event on this.once instead of doing `load()` at this point?

@katspaugh thoughts?

**Comment by katspaugh** [comment]
While I acknowledge the bug in #1414, breaking another thing to fix it is not a good solution.

The `interaction` event, AFAIR, is there for iOS devices. They require the user to interact with the page before being able to load audio. I think it's only valid for the HTML media, so I'm not sure why this event is subscribed to unconditionally.

**Comment by mustafa0x** [comment]
This bug wasted me several hours.

Can't @entonbiba's solution me applied?

**Comment by thijstriemstra** [comment]
> Can't @entonbiba's solution be applied?

Like @katspaugh said "breaking another thing to fix it is not a good solution."

**Review by entonbiba — COMMENTED** [review]
might be better to do "interaction" on ios devices only, and by default "ready". @katspaugh

File: src/wavesurfer.js [hunk 1]
```diff
@@ -1233,10 +1233,8 @@ export default class WaveSurfer extends util.Observer {
         if (peaks) {
             this.backend.setPeaks(peaks, duration);
             this.drawBuffer();
-            this.tmpEvents.push(this.once('interaction', load));
```
[the remark this answers is not in the document]
Comment (entonbiba) [reply src/wavesurfer.js:1236]: @sundayz @Guisch what you guys think about the above suggestion, you see any issues with the `this.once('ready', load)` ?

[the remark this answers is not in the document]
Comment (Guisch) [reply src/wavesurfer.js:1236]: Please see https://github.com/katspaugh/wavesurfer.js/issues/1414#issuecomment-768623044

**Comment by entonbiba** [comment]
closing this as pr #2176 fixes this for mobile devices

**Pull request closed without merging**